src/main/java/dev/lambdaurora/aurorasdeco/util/AuroraUtil.java

```diff
@@ -85,6 +89,21 @@ public static Identifier appendWithNamespace(String prefix, Identifier id) {
 		return AurorasDeco.id(prefix + '/' + path);
 	}
 
+	public static NbtCompound getOrCreateBlockEntityNbt(ItemStack stack, BlockEntityType<?> type) {
+		var nbt = BlockItem.getBlockEntityNbt(stack);
+		if (nbt == null) {
+			// setBlockEntityNbt only actually sets the nbt tag if it isn't empty.
+			// So we put in a dummy tag, because we actually want to *create* a new empty nbt tag.
```

Comment:
Not sure to understand that.

Reply:
It's a bit hacky.
In 1.17, you used `ItemStack.getOrCreateSubNbt(BlockItem.BLOCK_ENTITY_TAG_KEY)`. If the sub-NBT doesn't exist, this creates an empty one and returns it.
In 1.18, `BlockItem.BLOCK_ENTITY_TAG_KEY` became private, and instead the helper methods `BlockItem.getBlockEntityNbt` and `setBlockEntityNbt` got added. However, `setBlockEntityNbt` actually checks and refuses to add an empty tag, which is a problem because we want to put in an empty tag in case it doesn't exist.

My solution is to create an empty tag and add a dummy boolean to it *so that it's no longer empty*. This way, wenn I call  `setBlockEntityNbt`, it hits the code path for "non-empty tag" that actually adds the tag. I then remove the dummy boolean again because it's no longer needed.

I see three possible solutions:
 - what I've done so far. If you want, I can explain it more thoroughly in the comments.
 - use a mixin/accessor to effectively make `BlockItem.BLOCK_ENTITY_TAG_KEY` public, and do everything like in 1.17.
 - restructure the rest of the code to not use the "get the tag or insert an empty one if it doesn't exist" pattern. This might make the code more verbose less clear, because this pattern is actually quite useful and natural.